singularity-eos/eos/get_sg_eos_functors.hpp

```diff
@@ -216,12 +277,144 @@ struct final_functor {
     if (do_s) {
       sie_v(i) /= mass_sum;
     }
+    check_all_vals(i, npte, tid, pte_converged);
     /* reset mass fractions to original values if not normalized to 1 */
     for (int m = 0; m < nmat; ++m) {
       frac_mass_v(i, m) *= mass_sum;
     }
     return;
   }
+
+ private:
+  // Check to make sure returned values are normal or zero
+  template <typename valT>
+  PORTABLE_FORCEINLINE_FUNCTION int bad_val(valT const value,
+                                            const char *const var_name) const {
+    const bool good =
+        (value == valT{0} || (std::isnormal(1e10 * value) && std::isnormal(value)));
+    if (not good) {
+      using PortsOfCall::printf;
+      printf("### ERROR: Bad singularity-eos output\n  Var: %s\n  Value: %e\n", var_name,
+             value);
+    }
+    return not good;
+  }
+  // Check for prrecluding zero and negative (non-positive) values
+  template <typename valT>
+  PORTABLE_FORCEINLINE_FUNCTION int non_positive_val(valT const value,
+                                                     const char *const var_name) const {
+    const bool not_positive = value <= valT{0};
+    if (not_positive) {
+      using PortsOfCall::printf;
+      printf("### ERROR: Bad singularity-eos output\n  Var: %s\n  Value: %e\n", var_name,
+             value);
```

Comment:
Having the `using` just to bring `printf` into scope is a bit clunky. Either have `using PortsOfCall::printf;` at object or namespace scope, or just directly call `PortsOfCall::printf( /*...*/ );`

Reply:
I was intentionally limiting the scope that it was used in, but I suppose there's no reason to limit the scope below the object scope. I'll move that

Reply:
This now lives at the namespace level of the error utilities